cloud/openstack/os_quota.py

```diff
@@ -0,0 +1,339 @@
+    instances:
+        required: False
+        default: None
+        description: Maximum number of instances
+    key_pairs:
+        required: False
+        default: None
+        description: Number of key pairs to allow
+    network:
+        required: False
+        default: None
+        description: Number of networks to allow
+    per_volume_gigabytes:
+        required: False
+        default: None
+        description: Max size of individual volumes
+    port:
+        required: False
+        default: None
+        description: Number of Network ports to allow, should be greater than instances.
+    project:
+        required: False
+        default: None
+        description: Project name
+    properties:
+        required: False
+        default: None
+        description: Number of properties to allow
+    ram:
+        required: False
+        default: None
+        description: Maximum amount of ram
+    rbac_policy:
+        required: False
+        default: None
+        description: Number of polcies to allow
+    router:
+        required: False
+        default: None
+        description: Number of routers to allow
+    secgroup_rules:
+        required: False
+        default: None
+        description: Number of rules per security group to allow
+    secgroups:
+        required: False
+        default: None
+        description: Number of security groups to allow
+    server_group_members:
+        required: False
+        default: None
+        description: Number of server group members to allow
+    server_groups:
+        required: False
+        default: None
+        description: Number of server groups to allow
+    snapshots:
+        required: False
+        default: None
+        description: Number of snapshots to allow
+    snapshots_lvm:
+        required: False
+        default: None
+        description: Number of LVM snapshots to allow
+    subnet:
+        required: False
+        default: None
+        description: Number of subnets to allow
+    subnetpool:
+        required: False
+        default: None
+        description: Number of subnet pools to allow
+    volumes:
+        required: False
+        default: None
+        description: Number of volumes to allow
+    volumes_lvm:
+        required: False
+        default: None
+        description: Number of LVM volumes to allow
+
+
+requirements:
+    - "python >= 2.6"
+    - "shade > 1.9.0"
+'''
+
+
+EXAMPLES = '''
+# List a Project Quotas
+- os_quota:
+    cloud: mycloud
+    name: demoproject
+    cores: 100
+
+# Update a Project Quotas
+- os_quota:
+    name: demoproject
+    cores: 1000
+    volumes: 20
+    volumes_type:
+      - volume_lvm: 10
+'''
+
+
+RETURN = '''
+openstack_quotas:
+    description: Dictionary describing the quota.
+    returned:
+'''
+
+
+def exit_hostvars(module, cloud, project_quota, changed=True):
+    module.exit_json(changed=changed, ansible_facts=dict(
+        openstack_quotas=project_quota))
```

Comment:
This seems like a weird thing to turn into a function you call exactly once. Why make it a function?